prboom-plus: init at 2.6.2

kira64xyz opened the pull request:
###### Description of changes

This adds the popular PrBoom+ Doom source port (https://github.com/coelckers/prboom-plus)

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Commit:
prboom-plus: init at 2.6.2

IvarWithoutBones commented on pkgs/games/prboom-plus/default.nix:
```diff
@@ -0,0 +1,63 @@
+    SDL2_image
+    SDL2_net
+    fluidsynth
+    soundfont-fluid
```
Since we're directly using the soundfont's path below in `prePatch` we shouldn't need this in `buildInputs`, see this section of the [manual](https://nixos.org/manual/nixpkgs/unstable/#ssec-cross-dependency-implementation) for more information. These are essentially just used for linking, but since we need to do this manually it won't affect anything.

IvarWithoutBones reviewed (commented):
Thanks for the PR! This seems to be working perfectly :+1: 

Seeing how the [original prboom](https://github.com/NixOS/nixpkgs/blob/634141959076a8ab69ca2cca0f266852256d79ee/pkgs/games/prboom/default.nix) is both unmaintained, and there hasn't been a release [since 2008](http://prboom.sourceforge.net/), I think we should remove it from nixpkgs and replace it with this fork. You could remove the derivation and add a message explaining it's been removed to `pkgs/top-level/aliases.nix` in a seperate commit.

Also, could we maybe run unit tests with `doCheck = true;`? It looks like they do [exist upstream](https://github.com/coelckers/prboom-plus/tree/master/prboom2/tests).

Commit:
prboom: remove

PrBoom has not seen a release since 2008 and is essentially superseded
by PrBoom+

kira64xyz commented:
Thank you for the feedback!
I removed the original prboom and added an explanation to aliases.nix

Running the tests would be a lot of trouble as they require networking, so I opted to skip them for now

IvarWithoutBones commented:
@ofborg eval

IvarWithoutBones reviewed (approved):
Thank you for the PR, it's much appreciated. Already approved as well, I'll merge this now :blush: 

Result of `nixpkgs-review pr 170092` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>prboom-plus</li>
  </ul>
</details>
